src/lib/RequestParser.ts

```diff
@@ -19,27 +19,36 @@ import {
     ParsedRenameRequest,
     ParsedRpcRequest,
 } from './RequestTypes';
+import { Utf8Tools } from '@nimiq/utils';
 
 export class RequestParser {
     public static parse(request: RpcRequest, state: State, requestType: RequestType): ParsedRpcRequest | null {
         switch (requestType) {
             case RequestType.SIGN_TRANSACTION:
                 const signTransactionRequest = request as SignTransactionRequest;
+
+                if (!signTransactionRequest.value) throw new Error ('value is required');
+                if (!signTransactionRequest.validityStartHeight) throw new Error('validityStartHeight is required');
+
```

Comment:
there are more non-optional parameters to this request. Why check for `value` and `validityStartHeight` but not for the others? I think the `sender` and `recipient` will throw if not valid addresses, but `appName` and `accountId` should be checked.

Reply:
The `accountId` requirement was removed in #197.

Reply:
I see. Still leaves `appName`, which can not be undefined.

Reply:
https://github.com/nimiq/accounts/pull/195#discussion_r273036120